quickwit/quickwit-indexing/src/actors/doc_processor.rs

```diff
@@ -293,35 +334,64 @@ impl DocProcessor {
     }
 
     #[cfg(feature = "vrl")]
-    fn get_json_doc(&mut self, input_doc: InputDoc) -> Result<JsonObject, DocProcessorError> {
-        if let Some(vrl_program) = self.transform_opt.as_mut() {
-            let vrl_doc = input_doc.try_into_vrl_doc()?;
-            let transformed_vrl_doc = vrl_program.transform_doc(vrl_doc)?;
-            if let Ok(JsonValue::Object(json_doc)) = serde_json::to_value(transformed_vrl_doc) {
-                Ok(json_doc)
-            } else {
-                Err(DocProcessorError::ParsingError)
+    fn get_json_docs(
+        input_doc: InputDoc,
+        vrl_program_opt: Option<&mut VrlProgram>,
+    ) -> JsonDocIterator {
+        if let Some(vrl_program) = vrl_program_opt {
+            match input_doc
+                .try_into_vrl_doc()
```

Comment:
:/

Reply:
If you don't want the pipeline, the easy way to break those easily is to introduce another function.

```
fn get_json_doc_with_vrl(input_doc: InputDoc, vrl_program:  VrlProgram) -> Result<JsonDocIterator, _> {
   // ... the code just goes
  let blabla =erwer()?;
  let blabla =erwer()?;
  let blabla =erwer()?;
  let blabla =erwer()?;
  Ok(blabla)
}
```

I think the current code is ok though.